src/Storages/Distributed/DistributedAsyncInsertBatch.cpp

```diff
@@ -60,6 +61,21 @@ bool isSplittableErrorCode(int code, bool remote)
     ;
 }
 
+auto convertPartElementsToString(const std::vector<std::string> & files)
+{
+    std::vector<std::string> vec;
+    for (auto index_value : files | boost::adaptors::indexed())
```

Comment:
Looks like we can use `std::ranges::views::take(8)` instead (ranges is more adapated in the code now)